src/gabarit.erl

```diff
@@ -1,25 +1,97 @@
 %%%===================================================================
-%%% @doc
+%%% @author Maartz <maartz@protonmail.com>
+%%%
+%%% @doc Main interface for the Gabarit template management system.
+%%%
+%%% This module provides a public API for interacting with the Gabarit
+%%% template system, including:
+%%%  - Namespace management for organizing templates
+%%%  - Template creation, retrieval, and management
+%%%  - Template versioning and version control
+%%%  - Template rendering and formatting with context data
+%%%
+%%% Gabarit allows for efficient storage, versioning, and rendering of
+%%% templates with support for namespaces, compression, and version control.
 %%% @end
 %%%===================================================================
 -module(gabarit).
--export([new/1]).
--export([template/1]).
--export([format/2, format/3]).
--export([render/2, render/3]).
 
-%%--------------------------------------------------------------------
-%% @doc Create a new template from file present in default template
-%% path.
+-export([
+    create_namespace/2,
+    list_namespaces/0
+]).
+
+-export([
+    new/1,
+    new_template/1,
+    new_template/2,
+    add_template/3,
+    get_template/1,
+    get_template_version/2,
+    update_template/2,
+    compile_template/2,
+    list_templates/1,
+    list_versions/1,
+    rollback/2
+]).
+
+-export([
+    template/1,
+    format/2,
+    format/3,
+    render/2,
+    render/3
+]).
+
+%%====================================================================
+%% Namespace Management Functions
+%%====================================================================
+
+%% @doc Creates a new namespace with a specified template limit.
+%%
+%% @param Namespace The namespace identifier
+%% @param Limit Maximum number of templates allowed in the namespace
+%% @returns ok | {error, Reason}
+%% @end
+create_namespace(Namespace, Limit) ->
```

Comment:
can you create `create_namespace/1` with a default limit? Actually, it could even be better to be a map() containing a list of option (for a future usage).

```erlang
create_namespace(Namespace) ->
  create_namespace(Namespace, #{}).
  
 create_namespace(Namespace, Opts) ->
  Limit = maps:get(limit, Opts, 123),
  ...
```